OSDOCS-14033: Release Note for 4.15.49 (#92106)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.15.49
[id="ocp-4-15-49_{context}"]
=== RHSA-2025:3790 - {product-title} 4.15.49 bug fix and security update

Issued: 16 April 2025

> {product-title} release 4.15.49, which includes security updates, is now available. The list of bug fixes that are included in this update is documented in the link:https://access.redhat.com/errata/RHSA-2025:3055[RHSA-2025:3055] advisory. The RPM packages that are included in this update are provided by the link:https://access.redhat.com/errata/RHBA-2025:3057[RHBA-2025:3792] advisory.

Reviewer: ```suggestion
{product-title} release 4.15.49, which includes security updates, is now available. The list of bug fixes that are included in this update is documented in the link:https://access.redhat.com/errata/RHSA-2025:3790[RHSA-2025:3790] advisory. The RPM packages that are included in this update are provided by the link:https://access.redhat.com/errata/RHBA-2025:3057[RHBA-2025:3792] advisory.
```

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.49 --pullspecs
----

[id="ocp-4-15-49-bug-fixes_{context}"]
==== Bug fixes

> * Previously, an update to the {ibm-cloud-name} Cloud Internet Services (CIS) implementation impacted the upstream Terraform plugin. If you attempted to create an external-facing cluster on {ibm-cloud-name}, the following error occurred:

Reviewer: ```suggestion
* Previously, an update to the {ibm-cloud-name} Cloud Internet Services (CIS) implementation impacted the upstream Terraform plugin. If you attempted to create an externally-facing cluster on {ibm-cloud-name}, the following error occurred:
```

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.15.49
[id="ocp-4-15-49_{context}"]
=== RHSA-2025:3790 - {product-title} 4.15.49 bug fix and security update

Issued: 16 April 2025

{product-title} release 4.15.49, which includes security updates, is now available. The list of bug fixes that are included in this update is documented in the link:https://access.redhat.com/errata/RHSA-2025:3790 [RHSA-2025:3790] advisory. The RPM packages that are included in this update are provided by the link:https://access.redhat.com/errata/RHBA-2025:3057[RHBA-2025:3792] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.49 --pullspecs
----

[id="ocp-4-15-49-known-issues_{context}"]
==== Known issues

> * IPsec is not supported on Red Hat Enterprise Linux (RHEL) compute nodes because of a `libreswan` incompatiblility issue between a host and an `ovn-ipsec` container that exist in each compute node.  (link:https://issues.redhat.com/browse/OCPBUGS-36688 [*OCPBUGS-36688*]) 

Reviewer 2: ```suggestion
* IPsec is not supported on {op-system-base-full} compute nodes because of a `libreswan` incompatiblility issue between a host and an `ovn-ipsec` container that exist in each compute node.  (link:https://issues.redhat.com/browse/OCPBUGS-36688 [*OCPBUGS-36688*]) 
```

[id="ocp-4-15-49-bug-fixes_{context}"]

> ==== Bug fixes

Reviewer 3: The release note for https://issues.redhat.com/browse/OCPBUGS-50591 is missing.

![Screenshot From 2025-04-16 13-53-17](https://github.com/user-attachments/assets/23ca9a4d-7589-4dda-8966-1e69e9252cc0)

> * Previously, an update to the {ibm-cloud-name} Cloud Internet Services (CIS) implementation impacted the upstream Terraform plugin. If you attempted to create an externally-facing cluster on {ibm-cloud-name}, the following error occurred:

Reviewer 2: Pretty sure this was correct before (if you use "externally", there would be no hyphen, but also "external-facing" seems synonymous with "externally facing")
```suggestion
* Previously, an update to the {ibm-cloud-name} Cloud Internet Services (CIS) implementation impacted the upstream Terraform plugin. If you attempted to create an external-facing cluster on {ibm-cloud-name}, the following error occurred:
```

+
[source,terminal]
----
ERROR Error: Plugin did not respond
ERROR
ERROR with module.cis.ibm_cis_dns_record.kubernetes_api_internal[0],
ERROR on cis/main.tf line 27, in resource "ibm_cis_dns_record" "kubernetes_api_internal":
ERROR 27: resource "ibm_cis_dns_record" "kubernetes_api_internal"
----
+
With this release, you can use the installation program to create an external cluster on {product-title} without the plugin issue. (link:https://issues.redhat.com/browse/OCPBUGS-54367[OCPBUGS-54367]) 

> // Awaiting approval from Marco Braga for OCPBUGS-54353

Reviewer 2: Are the bugs with "Awaiting approval" approved? If they are approved, remove the comments. If they aren't is it too early to merge?

* Previously, when installing a cluster on {aws-first} in existing subnets that were located in edge zones, such as a Local Zone or a Wavelength Zone, the `kubernetes.io/cluster/<InfraID>:shared` tag was missing in the subnet resources of the edge zone. With this release, a fix ensures that all subnets that are used in the `install-config.yaml` configuration file have the required tag. (link:https://issues.redhat.com/browse/OCPBUGS-54353[*OCPBUGS-54353*]) 

// Awaiting approval from Marco Braga for OCPBUGS-53139

> * Previously, iSCSI and Fibre Channel devices attached by multipath did not resolve correctly when partitioned. This was caused by improper handling of multipath devices. With this release, a storage bug is fixed so that the partitioned multipath storage is now correctly recognized. (https://issues.redhat.com/browse/OCPBUGS-53139[*OCPBUGS-53139*]) 

Reviewer 2: ```suggestion
* Previously, iSCSI and Fibre Channel devices attached by multipath did not resolve correctly when partitioned. This was caused by improper handling of multipath devices. With this release, the partitioned multipath storage is now correctly recognized. (https://issues.redhat.com/browse/OCPBUGS-53139[*OCPBUGS-53139*]) 
```

> * Previously, the *Cluster Settings* page would not properly render during a cluster update if the `ClusterVersion` did not receive a `Completed` update. With this release, the *Cluster Setting* page properly renders even if the `ClusterVersion` has not received a `Completed` update. (link:https://issues.redhat.com/browse/OCPBUGS-53138[*OCPBUGS-53138*]) 

Reviewer 2: Is "the `ClusterVersion`" actually "the Cluster Version Operator (CVO)"? (check with an SME but I think this is probably the case)
```suggestion
* Previously, the *Cluster Settings* page would not properly render during a cluster update if the Cluster Version Operator (CVO) did not receive a `Completed` update. With this release, the *Cluster Setting* page properly renders even if the CVO has not received a `Completed` update. (link:https://issues.redhat.com/browse/OCPBUGS-53138[*OCPBUGS-53138*]) 
```

// Awaiting approval from Marco Braga for OCPBUGS-53055

> * Previously, the *Observe* section on the web console did not show items contributed from plugins unless certain flags related to the monitoring were set. However, these flags prevented other plugins, such as logging, distributed tracing, network observability, and so on, from adding items to the *Observe* section. With this release, the monitoring flags are removed so that other plugins can add items to the *Observe* section. (link:https://issues.redhat.com/browse/OCPBUGS-53055[*OCPBUGS-53055*]) 

Reviewer 2: ```suggestion
* Previously, the *Observe* section on the web console did not show items contributed from plugins unless certain flags related to monitoring were set. However, these flags prevented other plugins, such as logging, distributed tracing, network observability, and so on, from adding items to the *Observe* section. With this release, the monitoring flags are removed so that other plugins can add items to the *Observe* section. (link:https://issues.redhat.com/browse/OCPBUGS-53055[*OCPBUGS-53055*]) 
```

> * Previously, Previously, on a condition.Status, the ignition-server controller overloaded the Kubernetes agent server (KAS) by updating that condition with the same message in every reconcile loop. The updates caused an overload of the KAS. With this release, the controller checks the message and validates whether it is the existing message so that the KAS is not overloaded. (link:https://issues.redhat.com/browse/OCPBUGS-50867[OCPBUGS-50867]) 

Reviewer 2: Typo here. Also, what does "on a condition.Status" mean? I'm not sure it's clear to the audience (end user) when they might encounter this issue.
```suggestion
* Previously, on a condition.Status, the ignition-server controller overloaded the Kubernetes agent server (KAS) by updating that condition with the same message in every reconcile loop. The updates caused an overload of the KAS. With this release, the controller checks the message and validates whether it is the existing message so that the KAS is not overloaded. (link:https://issues.redhat.com/browse/OCPBUGS-50867[OCPBUGS-50867]) 
```

> * Previously, the ignition-server controller based on a condition.Status overloaded the KAS by updating that condition with the same message in every reconcile loop. The updates caused an overload of the KAS. With this release, the controller checks the message and validates whether it is the existing message so that the KAS is not overloaded. (link:https://issues.redhat.com/browse/OCPBUGS-50867[*OCPBUGS-50867*]) 

Reviewer 2: The first sentence here is weird. When I went to look at the source, I found this bug is backported from the original which had a different bug text:
```suggestion
* Previously, `IgnitionPayload` could not be generated because of a setting in the `ImageDigestMirrorSet` (IDMS) and `ImageContentSourcePolicy` (ICSP) resources. The setting caused the Kubernetes agent server to receive patch requests that continuously modified the `status.Condition` for `ignition-server` pods. With this release, the patch command does not reapply if the `status.Condition` for `ignition-server` pods is the same as the initial patch request.  (link:https://issues.redhat.com/browse/OCPBUGS-50867[*OCPBUGS-50867*]) 
```
I don't know which is better, but since this version has already been published it might make sense to use the original text?

Author: I suggested the original text to the SME and he responded by providing the "ignition-server controller" information.
